src/resolve/resolved-dns-dnssec.c

```diff
@@ -13,6 +13,14 @@
 #include "sort-util.h"
 #include "string-table.h"
 
+#if HAVE_OPENSSL
+#  pragma GCC diagnostic push
+#    pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(RSA*, RSA_free, NULL);
```

Comment:
Any specific reason to move these here instead of adding the pragma to the header where they were before?

Reply:
I should have mentioned that in the commit message. I wanted to do that in a separate step but then just did one commit… Those functions are only used in this file, and I don't think we should add any more uses. So it seems better to "localize" the ugly parts in that one file.

Reply:
These are used only in this file, and should not be used in newer code. I think moving them there is reasonable.

Reply:
Suspected as much, but wanted to double check